src/sentry/api/endpoints/organization_dashboard_details.py

```diff
@@ -44,3 +131,37 @@ def delete(self, request, organization, dashboard):
         dashboard.save()
 
         return self.respond(status=204)
+
+    def put(self, request, organization, dashboard):
+        """
+        Edit an Organization's Dashboard
+        ```````````````````````````````````
+
+        Edit an individual organization's dashboard as well as
+        bulk edits on widgets (i.e. rearranging widget order).
+
+        :pparam string organization_slug: the slug of the organization the
+                                          dashboard belongs to.
+        :pparam int dashboard_id: the id of the dashboard.
+        :param array widgets: the array of widgets (consisting of a widget id and the order)
+                            to be updated.
+        :auth: required
+        """
+        serializer = DashboardWithWidgetsSerializer(
+            data=request.DATA,
+            context={'dashboard_id': dashboard.id}
+        )
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+        data = serializer.object
+
+        title = data.get('title')
+        if title:
+            dashboard.update(title=data['title'])
+
+        widgets = data.get('widgets')
+        if widgets:
+            reorder_widgets(dashboard.id, widgets)
```

Comment:
Should this reorder and the dashboard update be a single database transaction? I think right now you could have concurrency issues if a customer were to send simultaneous requests they could create a race condition for widgets being reordered and both requests could fail with partial state saved.